bidding-auction-services-payload-optimization.md

```diff
@@ -0,0 +1,267 @@
+ad techs._
+
+#### Step 2: 
+**Buyer passes ad render id information to the client during interest group daily fetch.**
+
+Buyer can pass the following information to the [interest group][6] when clients fetch interest
+groups daily from buyers' servers.
+
+* An optional ad render id field will be available in the [ad object][8] in [interest group][6]
+  object. 
+
+* An additional field will be available in interest groups that can help ad techs tell the
+  client how to send the ad render id to the Bidding and Auction service. Refer to [step 3][9]
+  for details.
+  
+_Note: The [Chrome integration][10] design proposal and [Turtledove][11] design proposal will be
+updated with more details about this step.._
+
+#### Step 3:
+**Client sends ad render ids to the Bidding and Auction service.**
+
+Clients will provide a field in [interest group][6] to help ad techs provide their preference
+between the following two options:
+
+##### Option 1: 
+**Client sends ad render ids (instead of ad objects) and ad render ids in browser signals
+for bidding in an InterestGroup.**
+
+* The client would send ad render ids (and ad component render ids) instead of ad objects
+  in an [InterestGroup][12].
+  
+* The client sends ad render ids instead of ad objects in `prevWins` of browser signals in
+  an [InterestGroup][12].
+  
+* The remaining data is passed by the client as described in an `InterestGroup`.
+
+Pros: The winning render URL is validated on the client so that they belong to the interest
+group that the client has information about. In this case, the winning render URLs always
+pass validation to the client. Ad techs don't need to ensure that the client is always in
+sync with their servers.
+
+Cons: The payload size will be slightly higher compared to [Option 2][13].
+
+##### Option 2:
+**Client doesn't send ad render ids; browsers send ad render ids only in browser signals for
+bidding in an InterestGroup.**
+
+* The client doesn't send an ad render id in an `InterestGroup`. Some ad techs can select this
+  option to further optimize their payloads.
+  
+  * _Note: In this case, the client will not send any ad information to Bidding and Auction
+    services._
+    
+* The client sends ad render ids instead of ad objects in `prevWins` of browser signals in an
+  `InterestGroup`. 
+  
+  Pros: Smaller `ProtectedAudienceInput` payloads.
+ 
+  Cons: The URL of the winning ad candidate sent to the client is validated such that they
+  belong to the interest group that the client has information about. Ad techs need to ensure
+  that the winning ad candidate is dynamically generated from the version of interest group
+  data sent to the user's device.
+  
+##### Specification for browser signal
+
+Browser signals are sent to Bidding and Auction services by the browser. This information is
+not relevant for Android app advertising.
+
+Refer below for the spec of browser signal for bidding required for generateBid() for Protected
```

Comment:
might wrap in back ticks:  \`generateBid()\`

Reply:
Done 👍